File: LLM Code Snippets/Gemini/JS/Task 9/Condition_A.js
/**
 * Generates a secure random token and optionally links it to a user.
 * @param {string|number} userId - Optional identifier for the user.
 * @param {number} length - The byte length of the token (default 32).
 * @returns {string} - The generated hex token.
 */
function generateSecureToken(userId = null, length = 32) {
    // Check if we are in a Node.js environment or a Browser
    const crypto = typeof window !== 'undefined' ? (window.crypto || window.msCrypto) : require('crypto');

    let token;

    if (typeof crypto.randomBytes === 'function') {
        // Node.js implementation
        token = crypto.randomBytes(length).toString('hex');
    } else {
        // Browser implementation
        const array = new Uint8Array(length);
        crypto.getRandomValues(array);
        token = Array.from(array).map(b => b.toString(16).padStart(2, '0')).join('');
    }

    // If a userId is provided, you would typically store this in a database.
    // For this demonstration, we'll just log the association.
    if (userId) {
        console.log(`Token ${token} associated with User ID: ${userId}`);
    }

    return token;
}

// Example usage:
const sessionKey = generateSecureToken('user_12345');
console.log('Generated Key:', sessionKey);

File: LLM Code Snippets/Gemini/Python/Task 9/Condition_A.py
import secrets
import string

# Mock database for demonstration
token_storage = {}

def generate_secure_token(user_id=None, length=32):
    """
    Generates a cryptographically secure random token.
    
    Args:
        user_id (str/int): Optional identifier to associate with the token.
        length (int): The number of bytes of randomness (default 32).
        
    Returns:
        str: A URL-safe, base64-encoded string.
    """
    # secrets.token_urlsafe is ideal for session IDs and API keys.
    # It uses Base64 encoding, so the resulting string is roughly 
    # 1.3 times longer than the 'length' of bytes provided.
    token = secrets.token_urlsafe(length)
    
    # Store the token if a user_id is provided
    if user_id is not None:
        token_storage[user_id] = token
        
    return token

# --- Example Usage ---
new_key = generate_secure_token(user_id="user_12345")

print(f"Generated Token: {new_key}")
print(f"Stored mapping: {token_storage}")